OpenRA.Mods.Common/Widgets/Logic/MapChooserLogic.cs

```diff
@@ -200,11 +200,15 @@ void SetupGameModeDropdown(MapClassification tab, DropDownButtonWidget gameModeD
 
 		void EnumerateMaps(MapClassification tab, ScrollItemWidget template)
 		{
+			int playerCountFilter = -1;
+			int.TryParse(mapFilter, out playerCountFilter);
```

Comment:
`playerCountFilter` is set to 0 if the conversion fails, which introduces an edge case: If you have a map installed with no players (e.g. a map that you're working on in the editor) it will show up for all non-numeric search terms.

You can fix this using the following:
```csharp
int playerCountFilter;
if (!int.TryParse(mapFilter, out playerCountFilter))
	playerCountFilter = -1;
```